src/util/bip32.rs

```diff
@@ -180,13 +180,10 @@ impl FromStr for ChildNumber {
     type Err = Error;
 
     fn from_str(inp: &str) -> Result<ChildNumber, Error> {
-        Ok(match inp.chars().last().map_or(false, |l| l == '\'' || l == 'h') {
-            true => ChildNumber::from_hardened_idx(
-                inp[0..inp.len() - 1].parse().map_err(|_| Error::InvalidChildNumberFormat)?
-            )?,
-            false => ChildNumber::from_normal_idx(
-                inp.parse().map_err(|_| Error::InvalidChildNumberFormat)?
-            )?,
+        Ok(if inp.chars().last().map_or(false, |l| l == '\'' || l == 'h') {
```

Comment:
Seems to me this should be extracted into a small function to make it clear what's the purpose here (`is_hardened_path (inp)` or smth).

Reply:
Or just a variable. Variable names can add a lot of clarity.
```rust
let is_hardened = ...;
Ok(if is_hardened { ...
```

Reply:
Done